uO_scrape.py

```diff
@@ -137,13 +138,163 @@ def get_courses(link):
 		comp = comp.split(':', 1)[-1].strip()
 		comp = [x.strip() for x in comp.split('/')[-1].split(',')]
 		#getting the prerequisites from after the colon in the sentence
-		pre = pre.split(':', 1)[-1].strip()
-		#extracting dependencies from the prerequisite list
-		dep = extract_codes('.'.join([x for x in pre.split('.') if ('cannot' not in x) and ('Must' not in x)]))
-		dep = set(dep).difference({code})
-		courses.append([code, title, credits, desc, comp, pre, dep])
+		pre = Prereq(pre)
+		dep = None
+		# pre = pre.split(':', 1)[-1].strip()
+		# #extracting dependencies from the prerequisite list
+		# dep = extract_codes('.'.join([x for x in pre.split('.') if ('cannot' not in x) and ('Must' not in x)]))
+		# dep = set(dep).difference({code})
+		courses.append([code, title, credits, desc, comp, pre.prereqs, dep])
 	return pd.DataFrame(courses, columns = ['code', 'title', 'credits', 'desc', 'components', 'prerequisites', 'dependencies'])
 
+class Prereq:
+	'''
+	Object used to hold information about prereqs and other information that is provided in the courseblockextra section of the uOttawa catalogue
+	'''
+	credit_count_code = "YYY0000"
+	code_4U = "YYY0001"
+
+	course_code_pattern = r"[A-Z]{3}\s*[0-9]{4,5}[A-Za-z]{0,1}"
```

Comment:
Using `{3,4}` is definitely better because it accounts for STAT 5708, updated in `e338d94`.

As for including the pattern in `code_re` I think the best way would be to have a separate regex file as you suggested above. The easiest way would probably be a `.py` that just holds all the patterns. Since python is interpreted and not compiled this has all the benefits of a config file with none of the limitations. I'll move all the patterns to a separate file and add a commit to this PR (shouldn't take too long).